src/uu/install/src/install.rs

```diff
@@ -748,29 +751,70 @@ fn perform_backup(to: &Path, b: &Behavior) -> UResult<Option<PathBuf>> {
 /// Returns an empty Result or an error in case of failure.
 ///
 fn copy_file(from: &Path, to: &Path) -> UResult<()> {
+    // open the source file and get a file descriptor, making sure the source file exists
+    #[cfg(unix)]
+    let src = fs::OpenOptions::new().read(true).open(from);
+    #[cfg(not(unix))]
+    let src = File::open(from);
+    if let Err(e) = src {
+        show_error!(
+            "Failed to open source file {}. Error: {:?}",
+            from.display(),
+            e
+        );
+        return Err(InstallError::InstallFailed(from.to_path_buf(), to.to_path_buf(), e).into());
+    }
+    let mut src = src.unwrap();
+
     // fs::copy fails if destination is a invalid symlink.
```

Comment:
this is outdated as fs::copy itself is no longer used.

however, straced install from GNU coreutils and confirmed it also unlinks -- kind of surprised, i would expect creation of a temporary file and rename (space permitting). that is to say the comment can be removed altogether or replaced with a statement this lines up with gnu

Reply:
Updated the comments in the latest push.